backend/src/test/java/hanglog/trip/presentation/CommunityControllerTest.java

```diff
@@ -0,0 +1,167 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import hanglog.auth.domain.MemberTokens;
+import hanglog.city.domain.City;
+import hanglog.community.dto.response.CommunitySingleTripResponse;
+import hanglog.community.dto.response.CommunityTripsResponse;
+import hanglog.community.presentation.CommunityController;
+import hanglog.community.service.CommunityService;
+import hanglog.global.ControllerTest;
+import jakarta.servlet.http.Cookie;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.restdocs.AutoConfigureRestDocs;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.data.jpa.mapping.JpaMetamodelMappingContext;
+import org.springframework.restdocs.payload.JsonFieldType;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.ResultActions;
+
+
+@WebMvcTest(CommunityController.class)
+@MockBean(JpaMetamodelMappingContext.class)
+@AutoConfigureRestDocs
+class CommunityControllerTest extends ControllerTest {
+
+    private static final List<City> CITIES = List.of(PARIS, LONDON);
+    private static final MemberTokens MEMBER_TOKENS = new MemberTokens("refreshToken", "accessToken");
+    private static final Cookie COOKIE = new Cookie("refresh-token", MEMBER_TOKENS.getRefreshToken());
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private CommunityService communityService;
+
+    @BeforeEach
+    void setUp() {
+        given(refreshTokenRepository.existsByToken(any())).willReturn(true);
+        doNothing().when(jwtProvider).validateTokens(any());
+        given(jwtProvider.getSubject(any())).willReturn("1");
+    }
+
+    private ResultActions performGetRequest() throws Exception {
+        return mockMvc.perform(get("/community/trips")
+                .queryParam("page", "0")
+                .queryParam("size", "1")
+                .header(AUTHORIZATION, MEMBER_TOKENS.getAccessToken())
+                .cookie(COOKIE)
+                .contentType(APPLICATION_JSON));
+    }
+
+    @DisplayName("공개된 전체 여행 목록을 조회한다")
+    @Test
+    void getTripsByPage() throws Exception {
+        // given
+        when(communityService.getTripsByPage(any(), any()))
+                .thenReturn(
+                        new CommunityTripsResponse(
+                                List.of(CommunitySingleTripResponse.of(LONDON_TRIP, CITIES, true, 1L)),
+                                1L
+                        )
+                );
```

Comment:
```suggestion
        when(communityService.getTripsByPage(any(), any()))
                .thenReturn(new CommunityTripsResponse(
                        List.of(CommunitySingleTripResponse.of(LONDON_TRIP, CITIES, true, 1L)),
                        1L
                ));
```

여기도 이하 동문입니다. (그냥 의견이라는 뜻)

Reply:
[cbf2455](https://github.com/woowacourse-teams/2023-hang-log/pull/587/commits/cbf2455b4eff9fb3e879c002b406b53ea3945a15)